hasher-matcher-actioner/terraform/submit_events/main.tf

```diff
@@ -0,0 +1,185 @@
+    effect    = "Allow"
+    actions   = ["sqs:SendMessage"]
+    resources = [aws_sqs_queue.submit_event_queue.arn]
+    principals {
+      type        = "Service"
+      identifiers = ["sns.amazonaws.com"]
+    }
+    condition {
+      test     = "ArnEquals"
+      variable = "aws:SourceArn"
+      values   = [aws_sns_topic.submit_event_notification_topic.arn]
+    }
+  }
+}
+
+resource "aws_sqs_queue_policy" "submit_event_queue" {
+  queue_url = aws_sqs_queue.submit_event_queue.id
+  policy    = data.aws_iam_policy_document.submit_event_queue.json
+}
+
+
+resource "aws_lambda_function" "submit_event_handler" {
+  function_name = "${var.prefix}_submit_event_handler"
+  package_type  = "Image"
+  role          = aws_iam_role.submit_event_handler.arn
+  image_uri     = var.lambda_docker_info.uri
+  image_config {
+    command = [var.lambda_docker_info.commands.submit_event_handler]
+  }
+  timeout     = 300
+  memory_size = 512
+  environment {
+    variables = {
+      DYNAMODB_TABLE        = var.datastore.name
+      SUBMISSIONS_QUEUE_URL = var.submissions_queue.url
+    }
+  }
+  tags = merge(
+    var.additional_tags,
+    {
+      Name = "SubmitEventFunction"
+    }
+  )
+}
+
+resource "aws_cloudwatch_log_group" "submit_event_handler" {
+  name              = "/aws/lambda/${aws_lambda_function.submit_event_handler.function_name}"
+  retention_in_days = var.log_retention_in_days
+  tags = merge(
+    var.additional_tags,
+    {
+      Name = "SubmitEventLambdaLogGroup"
+    }
+  )
+}
+
+resource "aws_iam_role" "submit_event_handler" {
+  name_prefix        = "${var.prefix}_submit_event_handler"
+  assume_role_policy = data.aws_iam_policy_document.lambda_assume_role.json
+  tags = merge(
+    var.additional_tags,
+    {
+      Name = "SubmitEventLambdaRole"
+    }
+  )
+}
+
+data "aws_iam_policy_document" "submit_event_handler" {
+  statement {
+    effect    = "Allow"
+    actions   = ["dynamodb:GetItem", "dynamodb:Query", "dynamodb:Scan", "dynamodb:PutItem", "dynamodb:UpdateItem"]
+    resources = ["${var.datastore.arn}*"]
+  }
+
+  statement {
+    effect = "Allow"
+    actions = [
+      "logs:CreateLogStream",
+      "logs:PutLogEvents",
+      "logs:DescribeLogStreams"
+    ]
+    resources = ["${aws_cloudwatch_log_group.submit_event_handler.arn}:*"]
+  }
+
+  statement {
+    effect    = "Allow"
+    actions   = ["cloudwatch:GetMetricStatistics"]
+    resources = ["*"]
+  }
+
+  statement {
+    effect    = "Allow"
+    actions   = ["sqs:GetQueueAttributes", "sqs:ReceiveMessage", "sqs:DeleteMessage"]
+    resources = [aws_sqs_queue.submit_event_queue.arn]
+  }
+
+  statement {
+    effect    = "Allow"
+    actions   = ["sqs:SendMessage"]
+    resources = [var.submissions_queue.arn]
+  }
+  dynamic "statement" {
+    for_each = var.partner_image_buckets
+
+    content {
+      actions   = ["s3:GetObject"]
+      effect    = "Allow"
+      resources = ["${lookup(statement.value, "arn", null)}/*"]
+    }
+  }
+  # statement {
```

Comment:
Remove if the 'dyanmic' block is already working for you.

Reply:
sure. left it as example but good to remove.